ui/dashboards/src/components/Variables/VariableList.tsx

```diff
@@ -24,51 +24,197 @@ import {
+            <Button
+              onClick={() => {
+                props.onCancel();
+              }}
+            >
+              Cancel
+            </Button>
+            <Button
+              disabled={props.variableDefinitions === variableDefinitions}
+              variant="contained"
+              onClick={() => {
+                props.onChange(variableDefinitions);
+              }}
+            >
+              Apply Changes
+            </Button>
+          </Stack>
+          <Typography variant="h3" mb={2}>
+            Variable List
+          </Typography>
+          <Stack spacing={2}>
+            <TableContainer component={Paper}>
+              <Table sx={{ minWidth: 650 }} aria-label="simple table">
+                <TableHead>
+                  <TableRow>
+                    <TableCell>Visibility</TableCell>
+                    <TableCell>Variable Name</TableCell>
+                    <TableCell>Variable Type</TableCell>
+                    <TableCell align="right">Action</TableCell>
+                  </TableRow>
+                </TableHead>
+                <TableBody>
+                  {variableDefinitions.map((v, idx) => (
+                    <TableRow key={v.spec.name}>
+                      <TableCell component="th" scope="row">
+                        <Switch
+                          checked={v.spec.display?.hidden !== true}
+                          onChange={(e) => {
+                            toggleVariableVisibility(idx, e.target.checked);
+                          }}
+                        />
+                      </TableCell>
+                      <TableCell component="th" scope="row" sx={{ fontWeight: 'bold' }}>
+                        {v.spec.name}
+                      </TableCell>
+                      <TableCell>{v.kind}</TableCell>
+                      <TableCell align="right">
+                        <IconButton onClick={() => setVariableEditIdx(idx)}>
+                          <PencilIcon />
+                        </IconButton>
+                        <IconButton onClick={() => removeVariable(idx)}>
+                          <TrashIcon />
+                        </IconButton>
+                      </TableCell>
+                    </TableRow>
+                  ))}
+                </TableBody>
+              </Table>
+            </TableContainer>
+            <Box display="flex">
+              <Button onClick={addVariable} variant="contained">
+                Add New Variable
+              </Button>
+            </Box>
+          </Stack>
+        </>
+      )}
+    </Box>
+  );
+}
 
 export function TemplateVariableList() {
   const [isEditing, setIsEditing] = useState(false);
   const variableDefinitions = useTemplateVariableDefinitions();
   const { isEditMode } = useEditMode();
+  const [showVariablesInEditMode, setShowVariablesInEditMode] = useState(true);
+  const showVariables = isEditMode ? showVariablesInEditMode : true;
```

Comment:
do we need `showVariables`? seems like we can derive it from `showVariablesInEditMode`

Reply:
The value of showVariables is always true when it's not an editMode. The `showVariablesInEditMode` is only respected when its in edit mode

Reply:
oh if it's always true, then why do we need it?

Reply:
oh got it, it was confusing at first glance haha